File: client-app/src/app/models/userProfile.ts
export default interface UserProfile {
  userName: string;
  displayName: string;
  image?: string;
  bio?: string;
  photos?: Photo[];
  followers: number;
  followings: number;
  isFollowing: boolean;
}

export interface Photo {
  id: string;
  url: string;
  isMain: boolean;
}

File: Application/Profiles/UserProfile.cs
using Domain;

namespace Application.Profiles
{
    public class UserProfile
    {
        public string UserName { get; set; }
        public string DisplayName { get; set; }
        public string Bio { get; set; }
        public string Image { get; set; }
        public ICollection<Photo> Photos { get; set; }
        // Means that the currently logged-in user follows owner of this user profile
        public bool IsFollowing { get; set; }
        public int Followers { get; set; }
        public int Followings { get; set; }
    }
}